src/Framework/Telemetry/TaskExecutionStats.cs

```diff
@@ -5,11 +5,36 @@
 
 namespace Microsoft.Build.Framework;
 
-internal class TaskExecutionStats(TimeSpan cumulativeExecutionTime, short executionsCount, long totalMemoryConsumption)
+internal class TaskExecutionStats(TimeSpan cumulativeExecutionTime, int executionsCount, long totalMemoryConsumption)
 {
+    private TaskExecutionStats()
+        : this(TimeSpan.Zero, 0, 0)
+    { }
+
+    internal static TaskExecutionStats CreateEmpty()
+        => new();
+
+    /// <summary>
+    /// Total execution time of the task in all nodes for all projects.
+    /// </summary>
     public TimeSpan CumulativeExecutionTime { get; set; } = cumulativeExecutionTime;
+
+    /// <summary>
+    /// Total memory consumption (across all executions) in bytes.
+    /// </summary>
     public long TotalMemoryConsumption { get; set; } = totalMemoryConsumption;
-    public short ExecutionsCount { get; set; } = executionsCount;
+
+    /// <summary>
+    /// Total number of execution of the tasks in all nodes for all projects.
+    /// </summary>
+    public int ExecutionsCount { get; set; } = executionsCount;
+
+    internal void AddAnother(TaskExecutionStats another)
```

Comment:
nit: naming is a bit strange, could you be more specific and add short documentation?